src/shared-components/field/index.tsx

```diff
@@ -43,13 +59,13 @@ class Field extends React.Component {
   render() {
     const {
       children: inputChild,
-      disabled,
-      messages,
-      messagesType,
-      hideMessagesIcon,
-      hintMessage,
-      label,
-      labelFor,
+      disabled = false,
```

Comment:
I think this is how we can handle default Values for Class components. We may need to agree on a convention and standarize for all components. The end goal I think it is to mark the prop as optional if it's optional for future usage in curology

Reply:
Since this is just a class component that only uses render, should we just go ahead and convert it to a function component? We can still apply "static" properties to it when the component is just a function, too. It shouldn't break functionality.

As far as a convention: that would work for simple components that only destructure once, but what if a class has 15 methods on it? We'd probably want to define something like:

```
getProps = () => {
  const {
    somePropValue = true,
  } = this.props;

  return {
    somePropValue,
  }
}

render() {
  const { somePropValue } = this.getProps();
}
```

You could also look at Matt's `get injectedProps()` function in our main app for inspiration on how to work around this.

Reply:
Field refactored to FC.
ok so I have this idea when all components are converted, go to each one and standardise some things like defaultProps and convert to FC all than can be converted. I wanted to get an example of this convention so I applied it to `DialogModal` it worked fine and it even has spread props so it's a complete example. Can u review DialogModal so we have it as model for future work

Reply:
What you committed works fine. I honestly don't have super strong preferences, so I think it's better to survey the team and see (especially amongst those who have a harder time with TypeScript) what they might prefer. 

It seems like in addition to handling it via ES6 defaults in a separate method, we could also just define a separate TypeScript type, and cast any class components w/ `defaultProps` defined with that type, e.g.

```tsx

type DialogModalProps = {
  children: React.ReactNode;
  onCloseIconClick?: () => void | null;
  title?: string;
};


/**
 * Equivalent to:
 * type DialogModalPropsWithDefaults = {
 *   children: React.ReactNode;
 *   onCloseIconClick: () => void | null;
 *   title: string;
 * }
 */
type DialogModalPropsWithDefaults = Required<DialogModalProps>;

// We could also just inline `Required<DialogModalProps>` but some folks aren't familiar with the built-in utility methods.
getProps = () => this.props as DialogModalPropsWithDefaults;
```

Right now we have both the ES6 defaults *and* the `defaultProps` defined, which some people might find very confusing. 

I also think getting folks comfortable converting class components over to function components is key. The large majority of our lifecycle method usage is so simple that the mental model of how they map over to `useEffect` is *usually* straightforward. But some cases are not. In the case of `dialogModal`, it's a pretty simple conversion, and I think 99 times out of 100 it's better to just do the work to convert something over, rather than debate around that work, but that speaks more to my comfort with doing that work (especially in `radiance-ui`, with very sufficient test coverage) than the actual practicality of that ratio 😅 

Anyway, I encourage you to take the lead on this--perhaps draft up an RFC and share with the team?